### PR TITLE
feat(route): fix and update IEEE route

### PR DESCRIPTION
**Pull request by Derekmini**
## 该 PR 相关 Issue / Involved issue

Close #

## 完整路由地址 / Example for the proposed route(s)

```routes
/ieee/journal/78
/ieee/journal/78/recent
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [ ] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [x] 英文文档 EN
- [x] 全文获取 fulltext
  - [x] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note
* change to use cookieJar;
* remove html code in abstract;
* update note in the docs/journal.md and docs/en/journal.md

### CODE REVIEW

File: lib/v2/ieee/journal.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            item,
+        });
+    await Promise.all(
+        list.map((item) =>
+            ctx.cache.tryGet(item.link, async () => {
+                if (item.abstract !== '') {
+                    const response3 = await got(`${host}${item.link}`);
+                    const $3 = response3.body.match(/metadata=(.*);/)[1];
+                    const { abstract } = JSON.parse($3);
+                    item.abstract = abstract.replace(/<.*?>/g, '');
```
[the remark this answers is not in the document]
Comment (Derekmini) [reply lib/v2/ieee/journal.js:62]: just try to fix it, but I'm not understand the alert.

File: lib/v2/ieee/recent.js [hunk 2]
```diff
@@ -81,13 +74,10 @@
...
-                        method: 'get',
-                        url: host.concat(item.link),
-                    });
-                    const content = itemResponse.body.match(/metadata=(.*);/)[1];
-                    const { abstract } = JSON.parse(content);
-                    item.abstract = abstract;
+                    const response3 = await got(`${host}${item.link}`);
+                    const $3 = response3.body.match(/metadata=(.*);/)[1];
+                    const { abstract } = JSON.parse($3);
+                    item.abstract = abstract.replace(/<.*?>/g, '');
```
[the remark this answers is not in the document]
Comment (Derekmini) [reply lib/v2/ieee/recent.js:79]: just try to fix it, but I'm not understand the alert.

File: lib/v2/ieee/journal.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Derekmini) [reply lib/v2/ieee/journal.js:62]: Is it the problem of the replace? I just want to remove the html tag in the abstract. How to change it? $_&

**Review by TonyRL — COMMENTED** [review]
`item.description` of some articles are not showing for `/journal/:journal/recent` after visiting `/journal/:journal`

**Comment by Derekmini** [comment]
Yes, I'm aware of that. When I paste those two urls into my browser, I discover that the second one's `item.description` isn't very functional. However, the outcome is the same as what I had previously checked when I enter those two in the RSS software in the order "/journal/:journal" first and `journal/:journal/recent` second.

Although the description cache portion codes in those two files are identical, I still want to investigate the issue. When I try to access those two urls, I suspect that the website may be blocking my access at the time of runing the second code. Just suspect, 😭

Have you encountered this scenario before? I checked the relevant code to identify the issue, but I have no idea how to fix it.

**Comment by TonyRL** [comment]
use
```js
list = await Promise.all(...);
```
instead of
```js
await Promise.all(...);
```

**Comment by Derekmini** [comment]
@TonyRL 
I appreciate your time and your assistance. I apologize for the incorrect use of ‘await Promise.all(...);,’ however other scripts of mine utilize this code.

When I try to create the first script in the RSSHub, I don't understand the use of "promise.all," but I do recall that I obtained it from a script. As a result, I believe that every one of my scripts has this issue, and I need to fix them all in a new branch.

Another query: Is the use of "promise.all" there required? Can I get rid of it? I just read the RSSHub documentation, there is no "promise.all" code.

**Comment by TonyRL** [comment]
It depends.
If there's an API that provides all data at once, you do not need a `Promise`. If you need to use the response from an URL for each item, then using `Promise` is one of the most straightforward ways.

More info: 
[Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise)
[Promise.all()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all)

**Comment by Derekmini** [comment]
@TonyRL 
I appreciate your assistance and will figure things out. a pleasant night to you.

**Comment by MiaoCheng0517** [comment]
Thanks for your sharing. Your RSS routes are really useful for me to follow the current issue of a IEEE journal. I have a question here. Could I follow IEEE **early access articals** by you routes?

**Comment by Derekmini** [comment]
> Thanks for your sharing. Your RSS routes are really useful for me to follow the current issue of a IEEE journal. I have a question here. Could I follow IEEE **early access articals** by using your routes?

Thank you for your message, and I'm glad to hear that the RSS routes are helpful for you in following the current issue of an IEEE journal. Regarding your question about following IEEE early access articles using the routes, I actually had a similar thought at first. However, after some reflection, I realized that the logic behind RSS is to obtain the latest content, and the longer you subscribe, the more enriched the content you will receive. If you want to access previous content, it is not necessary to use the RSS, but rather through browsing method. I hope this answers your question!

**Comment by MiaoCheng0517** [comment]
Thank you for your quick reply. You said that it is unnecessary to access previous issues by RSS. IEEE update newest articals that are accepted and peer-reviewed, but may not be fully edited or assigned to a publication issue or volume. Early Access articles will appear on a future issue. I don't know whether your routes could follow these articals or not. I would appreciate it if you are willing to teach me or add this function.

**Comment by Derekmini** [comment]
> Thank you for your quick reply. You said that it is unnecessary to access **previous issues** by RSS. IEEE update newest articals that are accepted and peer-reviewed, but may not be fully edited or assigned to a publication issue or volume. Early Access articles will appear on **a future issue**. I don't know whether your routes could follow these articals or not. I would appreciate it if you are willing to teach me or add this function.

I'm sorry for misunderstanding your meaning. These types of articles are similar to online first publications, which I have not encountered before, so I haven't done this kind of work. However, these contents should all be based on the same operation. If you would like to participate in this type of work, you can do some preparation first: provide me with the website address for the Early Access articles of a journal, as well as the website address for the latest issue of that journal.

**Comment by MiaoCheng0517** [comment]
Take IEEE journal 59 for example.
The url for the Early Access articles: https://ieeexplore.ieee.org/xpl/tocresult.jsp?isnumber=4374138
The url for the latest issue: https://ieeexplore.ieee.org/xpl/mostRecentIssue.jsp?punumber=59
Thank you in advance for you kind help.

**Comment by Derekmini** [comment]
> Exactly! Early Access articles on IEEE are similar to online first publicaitons on CNKI. Take IEEE journal 59 for example. The url for the Early Access articles: https://ieeexplore.ieee.org/xpl/tocresult.jsp?isnumber=4374138 The url for the latest issue: https://ieeexplore.ieee.org/xpl/mostRecentIssue.jsp?punumber=59 Thank you in advance for your kind help.

I have received the links you provided and reviewed my original code. I agree that it is possible to modify my existing route to obtain the Early Access route.

But because the modification process involves packet capture work and I am currently busy with other tasks, I suggest that:

- Search and install a packet capture software;

- Visit the Early Access page in the browser, and retrieve the new got.post URL from the packet capture software;

- Modify the code at [/lib/v2/ieee/journal.js#L13](https://github.com/DIYgod/RSSHub/blob/f91e29db93305d8589149622da2c060539c0708a/lib/v2/ieee/journal.js#L13) and [/lib/v2/ieee/journal.js#L22-L31](https://github.com/DIYgod/RSSHub/blob/f91e29db93305d8589149622da2c060539c0708a/lib/v2/ieee/journal.js#L22-L31)

After adjusting the rest of the content accordingly, you should be able to obtain the Early Access route.

If you have any questions, please feel free to reply to me and I will be happy to assist you.

**Comment by MiaoCheng0517** [comment]
Thank you very much. I will try it. Wish you all the best.

**Pull request merged**